src/app/utils/MGPFallible.ts

```diff
@@ -21,6 +21,8 @@ export abstract class MGPFallible<T> {
 
     public abstract getReasonOr(value: string): string
 
+    public abstract and(other: MGPFallible<T>): MGPFallible<T>
```

Comment:
euh, le and est sensé prendre quelle reason ?
C'est zarbi vu comme ça fraté !

Reply:
si l'un des deux foire il prend whichever (le premier dans cette implémentation). 
Ça permet de chaîner facilement plusieurs conditions, mais c'est peut-être un peu trop fonctionnel-inspiré et pas full adapté à JS (car "other" a déjà été évalué, donc calcul potentiellement inutile.) Utilisé qu'à un seul endroit donc yé soupe-rime